transport/split/writer.go

```diff
@@ -36,32 +42,65 @@ var _ io.ReaderFrom = (*splitWriterReaderFrom)(nil)
 // A write will end right after byte index prefixBytes - 1, before a write starting at byte index prefixBytes.
 // For example, if you have a write of [0123456789] and prefixBytes = 3, you will get writes [012] and [3456789].
 // If the input writer is a [io.ReaderFrom], the output writer will be too.
-func NewWriter(writer io.Writer, prefixBytes int64) io.Writer {
-	sw := &splitWriter{writer, prefixBytes}
-	if rf, ok := writer.(io.ReaderFrom); ok {
-		return &splitWriterReaderFrom{sw, rf}
+// It's possible to enable multiple splits with the [AddSplitSequence] option, which adds count splits every skipBytes bytes.
+// Example:
+// prefixBytes = 1, AddSplitSequence(count=2, bytes=6)
+// Array of [0 1 3 2 4 5 6 7 8 9 10 11 12 13 14 15 16 ...] will become
+// [0] [1 2 3 4 5 6] [7 8 9 10 11 12] [13 14 15 16 ...]
+func NewWriter(writer io.Writer, prefixBytes int64, options ...Option) io.Writer {
+	sw := &splitWriter{writer: writer, nextSplitBytes: prefixBytes, remainingSplits: []repeatedSplit{}}
+	for _, option := range options {
+		option(sw)
+	}
+	if len(sw.remainingSplits) == 0 {
+		// TODO(fortuna): Support ReaderFrom for repeat split.
+		if rf, ok := writer.(io.ReaderFrom); ok {
+			return &splitWriterReaderFrom{sw, rf}
+		}
 	}
 	return sw
 }
 
+type Option func(w *splitWriter)
+
+// AddSplitSequence will add count splits, each of skipBytes length.
+func AddSplitSequence(count int, skipBytes int64) Option {
+	return func(w *splitWriter) {
+		if count > 0 {
+			w.remainingSplits = append(w.remainingSplits, repeatedSplit{count: count, bytes: skipBytes})
+		}
+	}
+}
+
 func (w *splitWriterReaderFrom) ReadFrom(source io.Reader) (int64, error) {
-	reader := io.MultiReader(io.LimitReader(source, w.prefixBytes), source)
+	reader := io.MultiReader(io.LimitReader(source, w.nextSplitBytes), source)
```

Comment:
> [ReadFrom reads data from r until EOF or error. The return value n is the number of bytes read. Any error except EOF encountered during the read is also returned. ](https://pkg.go.dev/io#ReaderFrom)

Might be a but: it looks to me that in this case only the first split will be tolerated.

Reply:
I'm not sure what bug you are referring to. I've added support for ReaderFrom, and the tests pass.

Reply:
> I've added support for ReaderFrom, and the tests pass.

This comment was basically about broken ReadFrom, resolved now